ocaml/xapi/repository.ml

```diff
@@ -735,43 +720,33 @@ let apply_updates' ~__context ~host ~updates_info ~livepatches ~acc_rpm_updates
         )
       ]
       ) ;
-  (* Evaluate recommended/pending guidances *)
-  let recommended_guidances, pending_guidances =
-    let guidances =
+  (* Evaluate guidances *)
+  let guidances =
+    let guidances' =
       (* EvacuateHost will be applied before applying updates *)
       eval_guidances ~updates_info ~updates:acc_rpm_updates ~kind:Recommended
         ~livepatches:successful_livepatches ~failed_livepatches
       |> List.filter (fun g -> g <> Guidance.EvacuateHost)
+      |> fun l -> merge_with_unapplied_guidances ~__context ~host ~guidances:l
     in
-    let guidances' =
-      merge_with_unapplied_guidances ~__context ~host ~kind:Recommended
-        ~guidances
-    in
+    GuidanceSet.assert_valid_guidances guidances' ;
```

Comment:
Is it safe to move the assert to a point were not all guidances have been added yet? In particular the ones affected by the failed livepatches

Reply:
Originally, the assert was only done on `recommended_guidances`, which didn't have that livepatch guidance. I don't know if it interacts badly for some reason. I suggest leaving this now, just in case.